src/elm/parameters.js

```diff
@@ -13,12 +13,13 @@ class ParameterDef extends Expression {
     // If context parameters contains the name, return value.
     if (ctx && ctx.parameters[this.name] !== undefined) {
       return ctx.parameters[this.name];
+      // If the parent context contains the name, return that
+    } else if (ctx.getParentParameter(this.name) !== undefined) {
+      const parentParam = ctx.getParentParameter(this.name);
+      return parentParam.default != null ? parentParam.default.execute(ctx) : parentParam;
```

Comment:
Wouldn't this only take the default parameter of the parent library if it is defined, instead of what was passed in to the engine? The code seems to work despite this though.

Reply:
My best guess as to why this works is that the first case in this `if` handles when the param is passed into the engine and not defined in the parent lib.

Reply:
But I am by no means a `cql-execution` expert

Reply:
Ha.  Yeah.  TBH, when I first saw this code, I thought the same thing as @hossenlopp.  And then came to the same conclusion as @okeefm (although I did not verify).  But it seems like that must be what is happening here.

Reply:
So `ctx.getParentParameter` seems to either return the value for that parameter OR a ParameterDef. That seems like an issue. In the debugger I have seen it return both. So I'm pretty confused about this whole thing.

Reply:
Hmm... can you tell under what conditions it returns each?

Reply:
I think that's why we have: `return parentParam.default != null ? parentParam.default.execute(ctx) : parentParam;`

In theory, every parameter definition should either have a passed in value or a default value (that's a CQL requirement -- no null parameter values).  If it's a passed in value, I think it would go through the first `if` or maybe to the "else" in the ternary on line 19.  If there is a default value, then it should return that default value per the ternary.  Which is probably why it works.

Reply:
I agree that the way parameters work right now is confusing and a little difficult to follow.  But I'm not sure that now is the time to fix that, w/ others wanting a fix by COB.  As far as I can tell, this PR works in the standard use cases, and if there is a scenario it does not work, I expect it also doesn't work on master -- so this likely is not a regression.  But given it fixes the issue at hand, and we haven't yet identified a use case where it does not work, I'd lean toward accepting it.